### PR TITLE
Remote search for select field component

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
Addresses #16

### CODE REVIEW

File: src/app/public/modules/select-field/select-field-picker.component.html [hunk 1]
```diff
@@ -9,7 +9,10 @@
           [data]="data"
           [selectedIds]="selectedIds"
           (selectedIdsChange)="onSelectedIdsChange($event)">
-          <sky-list-toolbar>
+          <sky-list-toolbar
+            [inMemorySearchEnabled]="context?.inMemorySearchEnabled"
+            (searchApplied)="onSearchApplied($event)"
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/select-field/select-field-picker.component.html]: 👆 Neither of these properties exist in `list-builder` today. We'll have to add them.
`inMemorySearchEnabled` stops a search from kicking off "my precious", which creates a redundant event. Without it, there's no way to be certain your own remote data won't get overwritten by list-builder's returned data.
`searchApplied` gets fired as soon as a user submits a search.

File: src/app/public/modules/select-field/select-field.component.ts [hunk 2]
```diff
@@ -103,6 +103,13 @@ export class SkySelectFieldComponent implements ControlValueAccessor, OnDestroy
   @Input()
   public multipleSelectOpenButtonText: string;
 
+  /**
+   * When `inMemorySearchEnabled` is `false`, it will circumvent the list-builder search function,
+   * allowing consumers to provide results from a remote source, by updating the `data` value.
+   */
+  @Input()
+  public inMemorySearchEnabled: boolean;
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/select-field/select-field.component.ts:112]: Is the default `true` then?

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
Nothing major is jumping out at me.

File: src/app/public/modules/select-field/select-field.component.ts (hunk 2 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/select-field/select-field.component.ts:112]: Yep. I'll update the jsdoc comment to reflect that.

File: src/app/public/modules/select-field/select-field.component.ts [hunk 3]
```diff
@@ -124,6 +132,12 @@ export class SkySelectFieldComponent implements ControlValueAccessor, OnDestroy
   @Output()
   public addNewRecordButtonClick = new EventEmitter<void>();
 
+  /**
+   * Fires when a search is sumbitted from the picker's toolbar.
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/select-field/select-field.component.ts]: 'sumbitted'

File: src/app/public/modules/select-field/select-field-picker.component.ts [hunk 4]
```diff
@@ -29,28 +45,47 @@ import {
...
 @Component({
   selector: 'sky-select-field-picker',
   templateUrl: './select-field-picker.component.html',
   styleUrls: ['./select-field-picker.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class SkySelectFieldPickerComponent implements OnInit, AfterContentInit, OnDestroy {
+
+  /**
+   * @internal Fires when a search is sumbitted from the picker's toolbar.
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/select-field/select-field-picker.component.ts]: 'sumbitted'

Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/select-field/select-field-picker.component.ts]: Also, I'd put `@internal` below the description, on its own line.

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
Works great! Just a few small things and we're good.

File: src/app/public/modules/select-field/select-field-picker.component.ts (hunk 4 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/select-field/select-field-picker.component.ts]: Done.

File: src/app/public/modules/select-field/select-field.component.ts (hunk 3 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/select-field/select-field.component.ts]: Fixed.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**